graphql footprint query

ETBassist opened the pull request:
- Install GraphQL
- Configure GraphiQL
- Define friendship and car types and inputs for GraphQL
- Add fetchUserCars for GraphQL
- Define car mutation for GraphQL
- Change createCar mutation field to car
- WIP fetchUserMonthFootprint GraphQL query  Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>
- Create fetchUserMonthFootprint graphQL query Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>
- Remove extra variable from test Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>

jdcarey128 committed:
Install GraphQL

Co-authored-by: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-authored-by: Eugene Theriault <45305677+ETBassist@users.noreply.github.com>
Co-authored-by: James Belta <41242546+JBelta@users.noreply.github.com>
Co-authored-by: Jonathan Wilson <53539820+Jonathan-M-Wilson@users.noreply.github.com>
Co-authored-by: Kiera Allen <46658858+KieraAllen@users.noreply.github.com>
Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>
Co-authored-by: Sean Steel <65255478+s-steel@users.noreply.github.com>

jdcarey128 committed:
Configure GraphiQL

Co-authored-by: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-authored-by: Eugene Theriault <45305677+ETBassist@users.noreply.github.com>
Co-authored-by: James Belta <41242546+JBelta@users.noreply.github.com>
Co-authored-by: Jonathan Wilson <53539820+Jonathan-M-Wilson@users.noreply.github.com>
Co-authored-by: Kiera Allen <46658858+KieraAllen@users.noreply.github.com>
Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>
Co-authored-by: Sean Steel <65255478+s-steel@users.noreply.github.com>

jdcarey128 committed:
Define friendship and car types and inputs for GraphQL

Co-authored-by: Jonathan Wilson <53539820+Jonathan-M-Wilson@users.noreply.github.com>

jdcarey128 committed:
Merge branch 'main' of github.com:Carbon-Knight/carbon-knight-back-end into graphql-setup

jdcarey128 committed:
Add fetchUserCars for GraphQL

jdcarey128 committed:
Define car mutation for GraphQL

Co-authored-by: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-authored-by: Eugene Theriault <45305677+ETBassist@users.noreply.github.com>
Co-authored-by: James Belta <41242546+JBelta@users.noreply.github.com>
Co-authored-by: Jonathan Wilson <53539820+Jonathan-M-Wilson@users.noreply.github.com>
Co-authored-by: Kiera Allen <46658858+KieraAllen@users.noreply.github.com>
Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>
Co-authored-by: Sean Steel <65255478+s-steel@users.noreply.github.com>

jdcarey128 committed:
Change createCar mutation field to car

ETBassist committed:
WIP fetchUserMonthFootprint GraphQL query
 Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>

ETBassist committed:
Create fetchUserMonthFootprint graphQL query
Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>

ETBassist committed:
Remove extra variable from test
Co-authored-by: Robert Basulto <63683549+Eternal-Flame085@users.noreply.github.com>

ETBassist committed:
Remove pry

jdcarey128 commented on app/graphql/queries/fetch_user_month_footprint.rb:
```diff
@@ -0,0 +1,13 @@
+module Queries
+  class FetchUserMonthFootprint < Queries::BaseQuery
+    type Types::CarMonthlyMileage, null: true
```
I think the name for this should be changed to FetchUserCarMonthFootprint in case we add future endpoints with flights, etc. A UserMonthFootprint could be an endpoint that aggregates all month footprint types.

ETBassist replied:
Good thought, just pushed this change up 👍

carson-jardine replied:
Noice

jdcarey128 commented on app/graphql/types/friendship_type.rb:
```diff
@@ -0,0 +1,8 @@
+module Types
```
Hopefully we get to this!

ETBassist replied:
🤞

jdcarey128 commented on app/graphql/types/query_type.rb:
```diff
@@ -0,0 +1,20 @@
+    # field <:query_name>, resolver: Queries::ClassName(defined in query_file.rb)
+    # resolver: executes the logic in the resolve method of the above class 
+
+    # TODO: remove me
```
We can remove the test_field at this point.

ETBassist replied:
DELETED

jdcarey128 reviewed (changes requested):
A couple small changes. See comments.

ETBassist committed:
Rename fetchUserMonthFootprint

Change fetchUserMonthFootprint to fetchUserCarMonthFootprint
Also moved GrapiQL into group :development, :test because that's where
it failed to deploy last time

ETBassist committed:
Remove example text from mutation_type.rb

carson-jardine commented on db/seeds.rb:
```diff
@@ -5,3 +5,16 @@
+    user_id: i + 1, 
+    make: 'subaru', 
+    model: 'forester', 
+    year: '20010', 
```
Can we change this to a standard year?

ETBassist replied:
lol done

carson-jardine reviewed (changes requested):
This looks great to me, just want to do one very very small refactor in the seeds before merging :)

ETBassist committed:
Change to a more standard year in seeds.rb

Co-authored-by: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
